xarray/coding/cftime_offsets.py

```diff
@@ -700,6 +710,8 @@ def to_cftime_datetime(date_str_or_date, calendar=None):
         return date
     elif isinstance(date_str_or_date, cftime.datetime):
         return date_str_or_date
+    elif isinstance(date_str_or_date, (datetime, pd.Timestamp)):
+        return cftime.DatetimeProlepticGregorian(*date_str_or_date.timetuple())
```

Comment:
This is so that we can pass  `start` and `end` dates of `cftime_range` and `date_range` as any datetime types. I think this makes sense for  `date_range` especially as it is kinda "agnostic" as to whether cftime is used or not. Thus one can provide bounds as python datetime object but still request something incompatible with numpy's datetime64.